client_authorization_test fails

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

client_authorization_test fails consistently

### Version

main

### Regression

- [X] The issue is a regression

### Expected behavior

Test passes

### Actual behavior

Test fails

### How to Reproduce?

At least since [May 16](https://github.com/keycloak/keycloak/actions/runs/9104361542), the client_authorization_test fails in `main`
Last known date when this test ran successfully was [May 13](https://github.com/keycloak/keycloak/actions/runs/9055245281)

### Anything else?

_No response_

The issue is due to invalid session encountered in a token which causes the user session not found in `names` endpoint of the ui-ext extension (`UIRealmsResource.getRealms()` method). The relevant stack trace is:

```
2024-05-20 11:40:22,587 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-13) Uncaught server error: java.lang.RuntimeException: Error while reading attributes from security token.
        at org.keycloak.authorization.common.KeycloakIdentity.<init>(KeycloakIdentity.java:255)
        at org.keycloak.services.resources.admin.permissions.MgmtPermissions.initIdentity(MgmtPermissions.java:105)
        at org.keycloak.services.resources.admin.permissions.MgmtPermissions.<init>(MgmtPermissions.java:96)
        at org.keycloak.services.resources.admin.permissions.AdminPermissions.realms(AdminPermissions.java:46)
        at org.keycloak.admin.ui.rest.UIRealmsResource.lambda$getRealms$0(UIRealmsResource.java:57)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:193)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:194)
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:212)
        at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1950)
        at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
        at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
        at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:161)
        at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
        at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
        at org.keycloak.utils.ClosingStream$ClosingIterator.hasNext(ClosingStream.java:293)
        at org.keycloak.utils.StreamsUtil.throwIfEmpty(StreamsUtil.java:55)
        at org.keycloak.admin.ui.rest.UIRealmsResource.getRealms(UIRealmsResource.java:66)
        at org.keycloak.admin.ui.rest.UIRealmsResource$quarkusrestinvoker$getRealms_ead7451714695748ac20e4aeece328e85603cece.invoke(Unknown Source)
        at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:147)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:582)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1570)
Caused by: java.lang.NullPointerException: Cannot invoke "org.keycloak.models.UserSessionModel.getUser()" because "userSession" is null
        at org.keycloak.authorization.common.KeycloakIdentity.getUserFromToken(KeycloakIdentity.java:304)
        at org.keycloak.authorization.common.KeycloakIdentity.<init>(KeycloakIdentity.java:242)
        ... 29 more
```

--- a/js/apps/admin-ui/cypress/e2e/client_authorization_test.spec.ts
+++ b/js/apps/admin-ui/cypress/e2e/client_authorization_test.spec.ts
@@ -195,7 +195,7 @@ describe("Client authentication subtab", () => {
     );
   });
 
-  describe.skip("Client authorization tab access for view-realm-authorization", () => {
+  describe("Client authorization tab access for view-realm-authorization", () => {
     const clientId = "realm-view-authz-client-" + uuid();
 
     beforeEach(async () => {
@@ -241,11 +241,11 @@ describe("Client authentication subtab", () => {
       loginPage.logIn("test-view-authz-user", "password");
       keycloakBefore();
 
-      sidebarPage
-        .waitForPageLoad()
-        .goToRealm("realm-view-authz")
-        .waitForPageLoad()
-        .goToClients();
+      sidebarPage.waitForPageLoad().goToRealm("realm-view-authz");
+
+      cy.reload();
+
+      sidebarPage.waitForPageLoad().goToClients();
 
       listingPage
         .searchItem(clientId, true, "realm-view-authz")

--- a/rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/UIRealmsResource.java
+++ b/rest/admin-ui-ext/src/main/java/org/keycloak/admin/ui/rest/UIRealmsResource.java
@@ -52,9 +52,10 @@ public class UIRealmsResource {
             )}
     )
     public Stream<RealmNameRepresentation> getRealms() {
+        final RealmsPermissionEvaluator eval = AdminPermissions.realms(session, auth.adminAuth());
+        
         Stream<RealmNameRepresentation> realms = session.realms().getRealmsStream()
                 .filter(realm -> {
-                    RealmsPermissionEvaluator eval = AdminPermissions.realms(session, auth.adminAuth());
                     return eval.canView(realm) || eval.isAdmin(realm);
                 })
                 .map((RealmModel realm) -> {

--- a/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java
+++ b/services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java
@@ -97,12 +97,17 @@ class MgmtPermissions implements AdminPermissionEvaluator, AdminPermissionManage
     }
 
     private void initIdentity(KeycloakSession session, AdminAuth auth) {
-        if (Constants.ADMIN_CLI_CLIENT_ID.equals(auth.getToken().getIssuedFor())
-                || Constants.ADMIN_CONSOLE_CLIENT_ID.equals(auth.getToken().getIssuedFor())) {
-            this.identity = new UserModelIdentity(auth.getRealm(), auth.getUser());
+        final String issuedFor = auth.getToken().getIssuedFor();
 
+        if (Constants.ADMIN_CLI_CLIENT_ID.equals(issuedFor) || Constants.ADMIN_CONSOLE_CLIENT_ID.equals(issuedFor)) {
+            this.identity = new UserModelIdentity(auth.getRealm(), auth.getUser());
         } else {
-            this.identity = new KeycloakIdentity(auth.getToken(), session);
+            ClientModel client  = session.clients().getClientByClientId(auth.getRealm(), issuedFor);
+            if (client != null && Boolean.parseBoolean(client.getAttribute(Constants.SECURITY_ADMIN_CONSOLE_ATTR))) {
+                this.identity = new UserModelIdentity(auth.getRealm(), auth.getUser());
+            } else {
+                this.identity = new KeycloakIdentity(auth.getToken(), session);
+            }
         }
     }
 

